In [1]:
import pandas as pd
import numpy as np

In [2]:
datos = pd.read_csv("Dataset.csv",sep=";")

In [3]:
datos.head()

age           job  marital  education default  balance housing loan  \
0   58    management  married   tertiary      no     2143     yes   no   
1   44    technician   single  secondary      no       29     yes   no   
2   33  entrepreneur  married  secondary      no        2     yes  yes   
3   47   blue-collar  married    unknown      no     1506     yes   no   
4   33       unknown   single    unknown      no        1      no   no   

   contact  day month  duration  campaign  pdays  previous poutcome   y  
0  unknown    5   may       261         1     -1         0  unknown  no  
1  unknown    5   may       151         1     -1         0  unknown  no  
2  unknown    5   may        76         1     -1         0  unknown  no  
3  unknown    5   may        92         1     -1         0  unknown  no  
4  unknown    5   may       198         1     -1         0  unknown  no

# Analisis General

## Exploración Inicial

In [4]:
print(datos.shape)

(20739, 17)


In [5]:
print(datos.head())

   age           job  marital  education default  balance housing loan  \
0   58    management  married   tertiary      no     2143     yes   no   
1   44    technician   single  secondary      no       29     yes   no   
2   33  entrepreneur  married  secondary      no        2     yes  yes   
3   47   blue-collar  married    unknown      no     1506     yes   no   
4   33       unknown   single    unknown      no        1      no   no   

   contact  day month  duration  campaign  pdays  previous poutcome   y  
0  unknown    5   may       261         1     -1         0  unknown  no  
1  unknown    5   may       151         1     -1         0  unknown  no  
2  unknown    5   may        76         1     -1         0  unknown  no  
3  unknown    5   may        92         1     -1         0  unknown  no  
4  unknown    5   may       198         1     -1         0  unknown  no  


In [6]:
print(datos.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20739 entries, 0 to 20738
Data columns (total 17 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   age        20739 non-null  int64 
 1   job        20739 non-null  object
 2   marital    20739 non-null  object
 3   education  20739 non-null  object
 4   default    20739 non-null  object
 5   balance    20739 non-null  int64 
 6   housing    20739 non-null  object
 7   loan       20739 non-null  object
 8   contact    20739 non-null  object
 9   day        20739 non-null  int64 
 10  month      20739 non-null  object
 11  duration   20739 non-null  int64 
 12  campaign   20739 non-null  int64 
 13  pdays      20739 non-null  int64 
 14  previous   20739 non-null  int64 
 15  poutcome   20739 non-null  object
 16  y          20739 non-null  object
dtypes: int64(7), object(10)
memory usage: 2.7+ MB
None


Parece no haber nulos, ya que todas las variables tienen 20739 valores, ahora estadisticas descriptivas de las variables númericas

In [7]:
print(datos.describe())

                age        balance           day      duration      campaign  \
count  20739.000000   20739.000000  20739.000000  20739.000000  20739.000000   
mean      41.095762    1148.639230     15.976469    259.027243      3.089541   
std       11.251527    4501.290387      8.588330    261.613962      3.660176   
min       20.000000   -8019.000000      1.000000  -1389.000000      1.000000   
25%       33.000000      30.000000      8.000000    103.000000      1.000000   
50%       40.000000     350.000000     15.000000    178.000000      2.000000   
75%       49.000000    1191.000000     23.000000    317.000000      3.000000   
max      530.000000  527532.000000     31.000000   3881.000000     63.000000   

              pdays      previous  
count  20739.000000  20739.000000  
mean      -0.990935      0.000241  
std        1.305461      0.034720  
min       -1.000000      0.000000  
25%       -1.000000      0.000000  
50%       -1.000000      0.000000  
75%       -1.000000      0.

Se puede ver algunas inconsistencias como el "age" maximo 530, numeros negativos en "duration". Se analizara detalladamente después.

In [8]:
# Revisa valores nulos
print(datos.isnull().sum())

age          0
job          0
marital      0
education    0
default      0
balance      0
housing      0
loan         0
contact      0
day          0
month        0
duration     0
campaign     0
pdays        0
previous     0
poutcome     0
y            0
dtype: int64
